libs/design-system/src/navigation/NavMenu.tsx

```diff
@@ -1,40 +1,117 @@
+import { NavLink as ReactNavLink } from 'react-router-dom';
+import { Group, Transition, Popover } from '@mantine/core';
 import useStyles from './NavMenu.styles';
+import usePopoverStyles from './NavLinkPopover.styles';
 
 interface INavMenuProps {
-  menuItems: {
-    icon: ReactNode;
-    label: string;
-    link: string;
-    testId?: string;
-    rightSide?: ReactNode;
-    condition?: boolean;
-  }[];
+  menuItems: IMenuItem[];
 }
 
-export function NavMenu({ menuItems }: INavMenuProps) {
-  const { classes } = useStyles();
+interface IMenuItem {
+  icon: ReactNode;
+  label: string;
+  link: string;
+  testId?: string;
+  rightSide?: ReactNode | { component: ReactNode; displayOnHover: boolean };
+  condition?: boolean;
+  tooltipLabel?: string;
+}
 
+export function NavMenu({ menuItems }: INavMenuProps) {
   return (
     <div>
       {menuItems
         .filter(({ condition = true }) => condition)
-        .map(({ icon, link, label, testId, rightSide }) => (
-          <NavLink
-            key={link}
-            to={link}
-            className={({ isActive }) => (isActive ? classes.linkActive : classes.link)}
-            data-test-id={testId}
-          >
-            <Group spacing={10}>
-              <div className={classes.linkIcon}>{icon}</div>
-              <div className={classes.linkLabel}>{label}</div>
-            </Group>
-            <div>{rightSide}</div>
-          </NavLink>
-        ))}
+        .map(({ icon, link, label, testId, rightSide, tooltipLabel }) => {
+          return (
+            <NavLink
+              link={link}
+              icon={icon}
+              label={label}
+              tooltipLabel={tooltipLabel}
+              rightSide={rightSide}
+              key={link}
+            />
+          );
+        })}
     </div>
   );
 }
+
+const NavLink = ({ rightSide, link, testId, icon, label, tooltipLabel }: IMenuItem) => {
+  const [popoverOpened, setPopoverOpened] = useState(false);
+  const [isHovered, setIsHovered] = useState<string>(null);
+  const { classes } = useStyles();
+
+  const withOnHover = (rightSide as { component: ReactNode; displayOnHover: boolean })?.displayOnHover || false;
+
+  return (
+    <ReactNavLink
+      to={link}
+      className={({ isActive }) => (isActive ? classes.linkActive : classes.link)}
+      data-test-id={testId}
+      onMouseEnter={() => setIsHovered(link)}
+      onMouseLeave={() => setIsHovered(null)}
+    >
+      <Group spacing={10}>
+        <div className={classes.linkIcon}>{icon}</div>
+        <div className={classes.linkLabel}>{label}</div>
+      </Group>
+      {!withOnHover && rightSide}
+
+      {withOnHover &&
+        rightSideWithHover({
+          rightSide: rightSide,
+          isHover: isHovered === link,
+          tooltipLabel: tooltipLabel,
+          popoverOpened: popoverOpened,
+          setPopoverOpened: setPopoverOpened,
+        })}
+    </ReactNavLink>
+  );
+};
+
+const rightSideWithHover = ({
```

Comment:
Please move this to the separate component or include it in the `NavLink` body, because it's anti-pattern to do it that way and using additionally the hooks inside `usePopoverStyles`.